pandas/tests/indexes/datetimes/test_construction.py

```diff
@@ -441,6 +441,34 @@ def test_000constructor_resolution(self):
 
         assert idx.nanosecond[0] == t1.nanosecond
 
+    def test_disallow_setting_tz(self):
+        # GH 3746
+        dti = DatetimeIndex(['2010'], tz='UTC')
+        with pytest.raises(ValueError):
+            dti.tz = pytz.timezone('US/Pacific')
+
+    @pytest.mark.parametrize('tz', [
+        None, 'America/Los_Angeles', pytz.timezone('America/Los_Angeles'),
```

Comment:
can you add a similar battery of these tests to Timestamp as well (disabled setting already is disabled, but see if we have a test for it).